app/models/user.rb

```diff
@@ -16,6 +16,26 @@ class User < ActiveRecord::Base
   has_many :magics
   has_many :universes
 
+  # as_json would try to print the password digest, which requires authentication
+  def as_json(options={})
+    excludes = [:password_digest, :old_password]
+    options = {} if options.nil?
+    options[:except] ||= excludes
+    super(options)
+  end
+
+  def to_json(options={})
```

Comment:
`as_json` creates a hash structure, which you then pass to `ActiveSupport::json.encode` to actually encode the object as a JSON string. `to_json` converts it straight to an escaped JSON string.